pkg/aws/awscontrolplane/admit_awscontrolplane.go

```diff
@@ -81,40 +81,94 @@ func (a *Admitter) Admit(request *v1beta1.AdmissionRequest) ([]admission.PatchOp
-	if awsControlPlaneCR.Spec.AvailabilityZones != nil || !aws.IsHAVersion(releaseVersion) {
+
+	// We only need to default if attributes are not set
+	if awsControlPlaneCR.Spec.AvailabilityZones != nil && awsControlPlaneCR.Spec.InstanceType != "" {
 		return result, nil
 	}
-	// Trigger defaulting of the master availability zones
-	a.Log("level", "debug", "message", fmt.Sprintf("AWSControlPlane %s AvailabilityZones is nil and will be defaulted", awsControlPlaneCR.ObjectMeta.Name))
-	numberOfAZs := aws.DefaultMasterReplicas
-	fetch := func() error {
-		ctx := context.Background()
-
-		// We try to fetch the G8sControlPlane CR.
-		g8sControlPlane := &infrastructurev1alpha2.G8sControlPlane{}
-		{
-			a.Log("level", "debug", "message", fmt.Sprintf("Fetching G8sControlPlane %s", awsControlPlaneCR.Name))
-			err := a.k8sClient.CtrlClient().Get(ctx,
-				types.NamespacedName{Name: awsControlPlaneCR.GetName(),
-					Namespace: awsControlPlaneCR.GetNamespace()},
-				g8sControlPlane)
+	if aws.IsHAVersion(releaseVersion) {
+		// Trigger defaulting of the master instance type
+		if awsControlPlaneCR.Spec.InstanceType == "" {
+			a.Log("level", "debug", "message", fmt.Sprintf("AWSControlPlane %s InstanceType is nil and will be defaulted", awsControlPlaneCR.ObjectMeta.Name))
+			patch := admission.PatchAdd("/spec/instanceType", aws.DefaultMasterInstanceType)
+			result = append(result, patch)
+		}
+		// Trigger defaulting of the master availability zones
+		if awsControlPlaneCR.Spec.AvailabilityZones == nil {
+			a.Log("level", "debug", "message", fmt.Sprintf("AWSControlPlane %s AvailabilityZones is nil and will be defaulted", awsControlPlaneCR.ObjectMeta.Name))
+			numberOfAZs := aws.DefaultMasterReplicas
+			fetch := func() error {
+				ctx := context.Background()
+
+				// We try to fetch the G8sControlPlane CR.
+				g8sControlPlane := &infrastructurev1alpha2.G8sControlPlane{}
+				{
+					a.Log("level", "debug", "message", fmt.Sprintf("Fetching G8sControlPlane %s", awsControlPlaneCR.Name))
+					err := a.k8sClient.CtrlClient().Get(ctx,
+						types.NamespacedName{Name: awsControlPlaneCR.GetName(),
+							Namespace: awsControlPlaneCR.GetNamespace()},
+						g8sControlPlane)
+					if err != nil {
+						return microerror.Maskf(aws.NotFoundError, "failed to fetch G8sControlplane: %v", err)
+					}
+				}
+				numberOfAZs = g8sControlPlane.Spec.Replicas
+				return nil
+			}
+			b := backoff.NewMaxRetries(3, 1*time.Second)
+			err = backoff.Retry(fetch, b)
 			if err != nil {
-				return microerror.Maskf(aws.NotFoundError, "failed to fetch G8sControlplane: %v", err)
+				a.Log("level", "debug", "message", fmt.Sprintf("No G8sControlPlane %s could be found", awsControlPlaneCR.Name))
 			}
+			// We default the AZs
+			defaultedAZs := a.getNavailabilityZones(numberOfAZs, a.validAvailabilityZones)
+			patch := admission.PatchAdd("/spec/availabilityZones", defaultedAZs)
+			result = append(result, patch)
+		}
+	} else {
```

Comment:
This is the defaulting for pre HA Versions. We need to look up the values from the AWSCluster